server/src/main/java/io/druid/server/namespace/NamespacedExtractionModule.java

```diff
@@ -0,0 +1,448 @@
+import io.druid.query.extraction.namespace.URIExtractionNamespace;
+import io.druid.server.initialization.ZkPathsConfig;
+import io.druid.server.namespace.cache.NamespaceExtractionCacheManager;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+import org.apache.curator.utils.ZKPaths;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+
+/**
+ *
+ */
+public class NamespacedExtractionModule implements DruidModule
+{
+  private static final String NAMESPACE_ENABLED_KEY = "enabledNamespaces";
+  private static final Logger log = new Logger(NamespacedExtractionModule.class);
+  private static final String TYPE_PREFIX = "druid.query.extraction.namespace.cache";
+  private final ConcurrentMap<String, Function<String, String>> fnCache = new ConcurrentHashMap<>();
+
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return ImmutableList.<Module>of(
+        new Module()
+        {
+          @Override
+          public String getModuleName()
+          {
+            return "DruidNamespacedExtractionModule";
+          }
+
+          @Override
+          public Version version()
+          {
+            return Version.unknownVersion();
+          }
+
+          @Override
+          public void setupModule(SetupContext context)
+          {
+            context.registerSubtypes(ExtractionNamespace.class);
+          }
+        }
+    );
+  }
+
+  @Override
+  public void configure(Binder binder)
+  {
+    JsonConfigProvider.bind(binder, TYPE_PREFIX, NamespaceExtractionCacheManager.class);
+
+    binder
+        .bind(ExtractionNamespaceFunctionFactory.class)
+        .annotatedWith(Names.named(JDBCExtractionNamespace.class.getCanonicalName()))
+        .to(JDBCExtractionNamespaceFunctionFactory.class)
+        .in(LazySingleton.class);
+    binder
+        .bind(ExtractionNamespaceFunctionFactory.class)
+        .annotatedWith(Names.named(URIExtractionNamespace.class.getCanonicalName()))
+        .to(URIExtractionNamespaceFunctionFactory.class)
+        .in(LazySingleton.class);
+
+    binder
+        .bind(Key.get(Object.class, Names.named(NAMESPACE_ENABLED_KEY)))
+        .toInstance(new Object());
+
+    LifecycleModule.register(binder, NamespacedKeeper.class);
+  }
+
+  public static boolean isNamespaceEnabled(Injector injector)
+  {
+    return null != injector.getExistingBinding(Key.get(Object.class, Names.named(NAMESPACE_ENABLED_KEY)));
```

Comment:
I do not understand what this is doing or why it is doing it.  Basically, it appears like you are checking to see if the `injector` passed in was built including this Module.  I'm not sure why that's a good test though...